source/jormungandr/jormungandr/scenarios/helper_classes/helper_utils.py

```diff
@@ -167,13 +156,15 @@ def _extend_journey(pt_journey, fallback_dp, fallback_period_extremity):
     pt_journey.durations.walking += aligned_fallback.journeys[0].durations.walking
     pt_journey.durations.bike += aligned_fallback.journeys[0].durations.bike
     pt_journey.durations.car += aligned_fallback.journeys[0].durations.car
+    pt_journey.durations.ridesharing += aligned_fallback.journeys[0].durations.ridesharing
```

Comment:
It would be more maintainable to iterate over a list of modes. 
Maybe something along those lines:

```python
for mode in list_off_all_modes: # bike, car, walking etc...
    aligned_duration = getattr(aligned_fallback.journeys[0].durations, mode)
    duration = getattr(pt_journey.durations, mode)
    setattr(pt_journey.durations, mode, duration + aligned_duration)
```

So that if we add a new mode later (eg. scooter), we don't have to remember to add a new line here.

Reply:
Thanks!
I did something that looks better, but not sure it's all you had in mind.